cmd/bio-rd/bgp.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/bio-routing/bio-rd/cmd/bio-rd/config"
+	bgpserver "github.com/bio-routing/bio-rd/protocols/bgp/server"
+	"github.com/bio-routing/bio-rd/routingtable"
+	"github.com/bio-routing/bio-rd/routingtable/filter"
+	"github.com/bio-routing/bio-rd/routingtable/vrf"
+)
+
+const (
+	DefaultReconnectInterval = time.Second * 15
+)
+
+type bgpConfigurator struct {
+	srv bgpserver.BGPServer
+}
+
+func (c *bgpConfigurator) configure(cfg *config.BGP) error {
+	c.deconfigureRemovedSessions(cfg)
+
+	err := c.reconfigureModifiedSessions(cfg)
+	if err != nil {
+		return fmt.Errorf("could not reconfigure session: %w", err)
+	}
+
+	err = c.configureNewSessions(cfg)
+	if err != nil {
+		return fmt.Errorf("could not configure session: %w", err)
+	}
+
+	return nil
+}
+
+func (c *bgpConfigurator) deconfigureRemovedSessions(cfg *config.BGP) {
+	for _, p := range c.srv.GetPeers() {
+		found := false
+		for _, g := range cfg.Groups {
+			for _, n := range g.Neighbors {
+				if n.PeerAddressIP == p.Addr() && p.VRF() == c.srv.GetDefaultVRF() {
```

Comment:
The `p.VRF() == c.srv.GetDefaultVRF()` bit surprises me a bit, is the intention to only support BGP sessions in the main VRF, or is this subject to change in further iterations? :grin:

Reply:
This code is nearly unmodified to the old main.go code, so it's subject to change in a further iteration.
The first step focused on readability of the existing code to set a starting point for the next steps.

Reply:
Cool, thanks for untangling this! <3